utils/utils.go

```diff
@@ -39,6 +39,16 @@ func ParseJSON(body []byte, i interface{}) {
 	json.Unmarshal(body, i)
 }
 
+// ParseJSONMap parses json byte slice into a generic key-value map
+func ParseJSONMap(body []byte) (map[string]*json.RawMessage, error) {
```

Comment:
I think it would be better to use the json `unmarshall` method in conjunction with a struct that explicitly defines the type, so that we can be more specific in the code about what we are expecting. If there is a missing part of the json that is being unmarshalled, then it won't fail, but it would instead have an empty or 0 value for that part of the struct. Thoughts?

Reply:
I wrote this function to give us greater flexibility in "half parsing" JSON objects. There are also some cases where we don't know the exact type, and can't create a struct to automatically unmarshal. This happens with the `widget.API` field which is an interface not a struct, so we need to do more work in the `widget.Widget`'s `UnmarshalJSON` function. I'm happy to work with a simpler solution if we can find a way to get the natural `json.Unmarshal` function to work in all cases

Reply:
Okay your idea worked really well. The widget unmarshalling is much less whack than before